advanced/Scripts/update.sh

```diff
@@ -35,25 +35,41 @@ source "/opt/pihole/COL_TABLE"
 
 GitCheckUpdateAvail() {
     local directory
+    local repo_name repo_url
     directory="${1}"
     curdir=$PWD
     cd "${directory}" || return
 
     # Fetch latest changes in this repo
     git fetch --quiet origin
 
-    # @ alone is a shortcut for HEAD. Older versions of git
-    # need @{0}
-    LOCAL="$(git rev-parse "@{0}")"
-
-    # The suffix @{upstream} to a branchname
-    # (short form <branchname>@{u}) refers
-    # to the branch that the branch specified
-    # by branchname is set to build on top of#
-    # (configured with branch.<name>.remote and
-    # branch.<name>.merge). A missing branchname
-    # defaults to the current one.
-    REMOTE="$(git rev-parse "@{upstream}")"
+    # Check current branch. If it is master, then check for the latest available tag instead of latest commit.
+    curBranch=$(git rev-parse --abbrev-ref HEAD)
+    if [[ "${curBranch}" == "master" ]]; then
+        # get the latest local tag
+        LOCAL=$(git describe --abbrev=0 --tags)
+        # get remote repo_name and URL
+        repo_name="$(git config --get remote.origin.url |  awk -F '[/.]' '{ print $6}')"
```

Comment:
This does not work on my own Pi-hole because I manually switched from `https` to `git` protocol:
```
$ git config --get remote.origin.url
git@github.com:pi-hole/pi-hole.git
```

```suggestion
        repo_name="$(basename "$(git rev-parse --show-toplevel)")"
```

another way could be (closer to yours) where `.git` is stripped from the end of the URL (if present)
```suggestion
        repo_name="$(basename -s .git $(git remote get-url origin))"
```

Reply:
And, just because you have `var="$()"` here and without quotes above: It is generally not needed as word splitting is not performed right-hand side of an assignment (see https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Shell-Parameters) (Section 3.5.4).